cfme/tests/integration/test_openldap_auth.py

```diff
@@ -12,13 +12,21 @@ def setup_first_provider():
     setup_a_provider(validate=True, check_existing=True)
 
 
-def test_openldap_auth(request, setup_first_provider, configure_openldap_auth_mode):
+def pytest_generate_tests(metafunc):
     data = cfme_data.get("openldap_test", {})
     if not data:
         pytest.skip("No openldap_test section in yaml")
     group = Group(description=data["group_name"], role="EvmRole-user")
-    request.addfinalizer(group.delete)
-    group.create()
+    metafunc.parametrize(
```

Comment:
We have recently switched from `metafunc.parametrize` to `testgen.parametrize` (from `utils.testgen`) throughout the codebase because we add some custom uncollecting. It will not affect non-provider tests such as this one, but we would like to keep the codebase consistent to avoid someone using this over `testgen.parametrize` somewhere where it could cause problems.